lib/srv/desktop/discovery.go

```diff
@@ -223,7 +223,9 @@ func (s *WindowsService) lookupDesktop(ctx context.Context, hostname string) ([]
 	// wait 5 seconds for the default resolver to return
```

Comment:
nit: Presuming my other comment about not needing `case <-s.cfg.Clock.After(5 * time.Second)` -- if we pass the timeout into `queryResolver`, this section of the code is a little easier to follow. Also, under the same presumption, I think we can get rid of both select's. So the whole function becomes:

```go
// lookupDesktop does a DNS lookup for the provided hostname.
// It checks using the default system resolver first, and falls
// back to the configured LDAP server if the system resolver fails.
func (s *WindowsService) lookupDesktop(ctx context.Context, hostname string) ([]string, error) {
	stringAddrs := func(addrs []netip.Addr) []string {
		result := make([]string, 0, len(addrs))
		for _, addr := range addrs {
			result = append(result, addr.String())
		}
		return result
	}

	queryResolver := func(resolver *net.Resolver, resolverName string, timeout time.Duration) chan []netip.Addr {
		ch := make(chan []netip.Addr, 1)
		if resolver != nil {
			go func() {
				tctx, cancel := context.WithTimeout(ctx, timeout)
				defer cancel()

				addrs, err := resolver.LookupNetIP(tctx, "ip4", hostname)
				if err != nil {
					s.cfg.Log.Debugf("DNS lookup for %v failed with %s resolver: %v",
						hostname, resolverName, err)
				}

				ch <- addrs

			}()
		}
		return ch
	}

	// kick off both DNS queries in parallel, with timeout
	defaultResult := queryResolver(net.DefaultResolver, "default", dnsQueryTimeout)
	ldapResult := queryResolver(s.dnsResolver, "LDAP", dnsQueryTimeout)

	// wait dnsQueryTimeout for the default resolver to return
	addrs := <-defaultResult
	if len(addrs) > 0 {
		return stringAddrs(addrs), nil
	}

	// If we didn't get a result from the default resolver,
	// the result from the LDAP resolver is either available
	// now or we're done. There's no more waiting.
	addrs = <-ldapResult
	if len(addrs) > 0 {
		return stringAddrs(addrs), nil
	}

	return nil, trace.Errorf("could not resolve %v in time", hostname)
}
```